ott/core/layers.py

```diff
@@ -49,16 +49,6 @@ class PositiveDense(nn.Module):
                         Array] = nn.initializers.lecun_normal()
   bias_init: Callable[[PRNGKey, Shape, Dtype], Array] = nn.initializers.zeros
 
-  def setup(self):
-    try:
```

Comment:
Yes, with the `jax==0.3.18`, the `except` was no longer caught. I removed it because this check was brittle (and hence it broken), tried other approached like `checkify`, but it breaks the API.
The cleanest way would be to write an interpreter to invert any Jaxpr, as done [here](https://jax.readthedocs.io/en/latest/notebooks/Writing_custom_interpreters_in_Jax.html#custom-inverse-jaxpr-interpreter), but feel like it's too much work.

Reply:
ahh nice. indeed, I think it's overkill, let's keep the burden on the user.

As a side note, there's a `log(exp(x)-1)` as the inverse of the softplus that's really ugly. Couldn't find a better way (and not really worried, because that inverse is only used for small values, to ensure proper normalization) but I guess worth adding a todo

Reply:
I looked at TF, they seem to define it exactly as `log(exp(x)-1` [here](https://www.tensorflow.org/probability/api_docs/python/tfp/math/softplus_inverse).